pkg/kyc/component.go

```diff
@@ -118,33 +123,78 @@ func (c *component) GetUserByUsernameInSocialRealm(ctx context.Context, username
 		return apikyc.UserRepresentation{}, err
 	}
 
+	var realmConf configuration.RealmAdminConfiguration
+	if realmConf, err = c.configDBModule.GetAdminConfiguration(ctx, c.socialRealmName); err != nil {
+		return apikyc.UserRepresentation{}, err
+	}
+	var consentRequired = realmConf.ConsentRequired == nil || *realmConf.ConsentRequired
+
 	var kcUser keycloak.UserRepresentation
 	kcUser, err = c.getUserByUsername(accessToken, c.socialRealmName, c.socialRealmName, username, *group.ID)
 	if err != nil {
 		c.logger.Info(ctx, "msg", "GetUser: can't find user in Keycloak", "err", err.Error())
 		return apikyc.UserRepresentation{}, err
 	}
 	keycloakb.ConvertLegacyAttribute(&kcUser)
-	return c.getUser(ctx, *kcUser.ID, kcUser)
+
+	return c.getUser(ctx, *kcUser.ID, kcUser, !consentRequired)
+}
+
+func (c *component) createConsentError(errorType string) error {
+	return errorhandler.Error{
+		Status:  430,
+		Message: fmt.Sprintf("%s.%s.consent", errorhandler.GetEmitter(), errorType),
+	}
+}
+
+func (c *component) checkUserConsent(ctx context.Context, accessToken string, realm string, userID string, consentCode *string) error {
+	var rac, err = c.configDBModule.GetAdminConfiguration(ctx, realm)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Can't get realm admin configuration", "realm", realm, "err", err.Error())
+		return err
+	}
+
+	if rac.ConsentRequired == nil || *rac.ConsentRequired {
```

Comment:
for retro compatibility, ConsentRequired == nil might means false. What do you think of it ?

Reply:
We told it should be activated by default during our daily

Reply:
Done